1.11 release post. (#6541)
vercel/turbo: docs/pages/blog/turbo-1-11-0.mdx

---
title: Turborepo 1.11
date: 2023/11/27
description: Turborepo 1.11 sets the Rust-based foundation for the future of Turborepo.
tag: "web development"
ogImage: /images/blog/turbo-1-11-0/x-card.png
---

# Turborepo 1.11

import { Authors } from '../../components/Authors'
import { ExperimentalBadge } from '../../components/ExperimentalBadge'
import ThemeAwareImage from '../../components/ThemeAwareImage'
import Date from "../../components/blog/Date";

<Date>
  Monday, November 27th, 2023
</Date>

<Authors authors={[
  "gregsoltis",
  "chrisolszewski",
  "nicholasyang",
  "alexanderlyon",
]} />

Turborepo 1.11 completes our migration to Rust and ships several developer experience improvements:

- [**Our new Rust foundation**](#unlocking-the-future-of-turbo-with-rust): We've finished the port from Go to lay the groundwork for better performance, improved stability, and new features.
- [**Group logs for improved readability**](#group-logs-for-improved-readability): You can now specify `--log-order=grouped` to organize your logs into distinct sections separated by tasks.

> - [**Updated examples**](#updated-examples): Start with an example for Next.js, Svelte, Remix, Nuxt, and more.

Reviewer: These don't match the OG, are we including LSP?

Update today by running `npx @turbo/codemod migrate`.

## Unlocking the future of `turbo` with Rust

As we wrote about in [our blog post about the migration](https://vercel.com/blog/turborepo-migration-go-rust), we're excited to start seeing the benefits of our porting effort for Turborepo users. Rust has allowed us to surface more complexity in _our_ code so we're able to create a better experience in _your_ codebase in several ways:

> - We've noticed a slight performance improvement for the majority of use cases, even without doing much performance tuning in our code yet. We look forward to diving deeper into making `turbo` even faster soon.

Reviewer: Should we add some metrics? Feels pretty vague here. How much, and which use cases? Even naming one or two would be helpful in my opinion.

- We've been able to catch possible bugs much sooner, as Rust forces us to think about the differences between platforms and environments much more closely than Go.
- The Rust ecosystem has many powerful crates that served our needs well, showcasing the ecosystem alignment we were lacking in Go.

> If you find any problems that you believe are related to the new Rust codepath, you can opt back to the Go codepath using the `--go-fallback` flag and [report an issue on GitHub](https://github.com/vercel/turbo/issues/new?assignees=&labels=kind%3A+bug%2Cowned-by%3A+turborepo%2Cneeds%3A+triage&projects=&template=0-turborepo-bug-report.yml&title=%5Bturborepo%5D+).

Reviewer: Non blocking - we could create a label `rust-port` (or something) that gets auto added when using this link (set it here via query param).

## Group logs for improved readability

To make it easier to review your task logs, we've introduced the `--log-order` flag to output all logs from a task **after it finishes** rather than _as it writes_. This results in logs that are separated per-task, improving readability on a specific task's logs.

<ThemeAwareImage
  className="my-4 flex justify-center"
  light={{
    alt: 'Using `--log-order=grouped` to output task logs to the terminal as separated chunks separated by task.',
    src: '/images/blog/turbo-1-11-0/log-order-groups-light.png',
    props: {
      width: 682,
      height: 746.5,
      className: "dark:hidden block rounded-xl"
    }
  }}
  dark={{
    alt: 'Using `--log-order=grouped` to output task logs to the terminal as separated chunks separated by task.',
    src: '/images/blog/turbo-1-11-0/log-order-groups-dark.png',
    props: {
      width: 682,
      height: 746.5,
      className: "hidden dark:block rounded-xl"
    }
  }}
/>

> Enable this behavior using `--log-order=grouped`. [Check out the documentation](https://turbo.build/repo/docs/reference/command-line-reference/run#--log-order) to learn more

Reviewer: Internal links should be relative:

```suggestion
Enable this behavior using `--log-order=grouped`. [Check out the documentation](/repo/docs/reference/command-line-reference/run#--log-order) to learn more
```

## Updated examples

Our repository contains [`an examples directory`](https://github.com/vercel/turbo/tree/main/examples) with starter monorepos for your projects.

To get started with an example, run:

```bash

> # Uses the with-basic example

Reviewer: ```suggestion
# Next.js
```

npx create-turbo@latest
```

```bash
# Next.js and Remix
npx create-turbo@latest -e kitchen-sink
```

```bash
# Vue and Nuxt
npx create-turbo@latest -e with-vue-nuxt
```

We're putting a renewed effort towards ensuring our examples are kept up-to-date. Additionally, we've re-architected our examples to advocate for best practices in monorepos. For example:

- In-editor go-to-definition across packages
- Dependencies updated to latest versions
- Tooling like ESLint, Prettier, jest, and vitest pre-configured
- Hot-reloading across packages

> Check out [the examples directory](https://github.com/vercel/turbo/tree/main/examples) for more.

Reviewer: ```suggestion
[Choose an example](/repo/docs/getting-started/from-example) to get started, or [browse the code](https://github.com/vercel/turbo/tree/main/examples) directly for more.
```
